### PR TITLE
Cross-build paradise for 2.12.

### PR DESCRIPTION
**Pull request by olafurpg**
- This commit adds 2.12 support for scalameta/paradise macro annotations.
- 2.11/2.12 specific code is put into 2.11.8/2.12.1 directories, using
  the same convention as scalahost.

### CODE REVIEW

File: plugin/src/main/scala-2.12.1/org/scalameta/paradise/typechecker/Namers.scala [hunk 1]
```diff
@@ -0,0 +1,696 @@
+package org.scalameta.paradise
```
Comment (xeno-by) [inline plugin/src/main/scala-2.12.1/org/scalameta/paradise/typechecker/Namers.scala]: So, the only difference between paradise 2.11 and 2.12 is `Namers.scala`... Fascinating!

File: bin/bintray.sh [hunk 2]
```diff
@@ -8,7 +8,7 @@ host = api.bintray.com
 user = $BINTRAY_USERNAME
 password = $BINTRAY_API_KEY
 EOF
-  sbt -Dsbt.ivy.home=/drone/cache/ivy2 publish
+  sbt -Dsbt.ivy.home=/drone/cache/ivy2 +publish
 fi
```
Comment (olafurpg) [inline bin/bintray.sh]: That will require sbt-doge, which is probably a good idea anyways.

File: build.sbt [hunk 3]
```diff
@@ -19,6 +19,11 @@ lazy val paradiseRoot = Project(
     base = file(".")
   ) settings (
     sharedSettings,
+    commands += Command.command("ci") { state =>
+    ScalaVersions.foldRight(state) {
+      case (v, s) => s"++$v test" :: s
+    }
+  },
```
Comment (olafurpg) [inline build.sbt]: With sbt-doge it can be the same.

It's a known scalafmt bug (https://github.com/olafurpg/scalafmt/issues/701#issue-203459084), I should really fix it

File: build.sbt [hunk 4]
```diff
@@ -98,11 +104,19 @@ lazy val sharedSettings = Def.settings(
...
+    // not "scala_2.11.8" or "scala_2.12.1" that we need.
+    // That's why we have to work around here.
+    val base = (sourceDirectory in Compile).value
+    base / ("scala-" + scalaVersion.value)
+  },
   triggeredMessage in ThisBuild := Watched.clearWhenTriggered
 )
 
 lazy val mergeSettings = Def.settings(
-  sharedSettings,
```
Comment (olafurpg) [inline build.sbt]: Because it's redundant and scala-2.11.8 appeared twice everywhere in unmanagedSourceDirectories.

File: tests/common/src/main/scala/ToolSuite.scala [hunk 5]
```diff
@@ -22,6 +22,8 @@ class ToolSuite(project: String) extends FunSuite {
     } catch {
       case SystemExitException(exitCode) =>
         outputStream.close(); (exitCode, outputStorage.toString)
```
Comment (olafurpg) [inline tests/common/src/main/scala/ToolSuite.scala]: Nope, this is a custom exception used in Scaladoc via `checkExit`. I'm sure this could be cleaned up better somehow, but I'd prefer not to block this PR by it.

**Comment by xeno-by** [comment]
Thank you very much for your work! I've been going through the pain of accommodating small changes in compiler internals for 3+ years, and it's quite tedious. I really appreciate your help in this area.

File: plugin/src/main/scala/org/scalameta/paradise/reflect/Metadata.scala [hunk 6]
```diff
@@ -49,16 +49,16 @@ trait Metadata { self: ReflectToolkit =>
   }
 
   class Metadata[T: Attachable](carrier: T) {
-    import scala.collection.JavaConversions._
+    import scala.collection.JavaConverters._
```
Comment (olafurpg) [inline plugin/src/main/scala/org/scalameta/paradise/reflect/Metadata.scala]: I don't really understand how Metadata is used, so I think it might take me a while to consolidate. I'd prefer to do it in a separate PR, I opened https://github.com/scalameta/paradise/issues/169

**Comment by xeno-by** [comment]
Superseded by https://github.com/scalameta/paradise/pull/170
